src/Core/System/StateMachine/StateMachineRegistry.php

```diff
@@ -257,8 +257,8 @@ private function getTransitionDestinationById(string $stateMachineName, string $
         foreach ($stateMachineTransitions as $transition) {
             /** @var StateMachineStateEntity $toState */
             $toState = $transition->getToStateMachineState();
-            // Always allow to cancel a payment whether its a valid transition or not
-            if ($transition->getActionName() === 'cancel' && $transitionName === 'cancel') {
+            // Always allow to cancel a payment whether it's a valid transition or not
+            if ($stateMachineName === 'order_transaction.state' && $transition->getActionName() === 'cancel' && $transitionName === 'cancel') {
```

Comment:
Can you please add a test case in `\Shopware\Tests\Integration\Core\System\StateMachine\StateMachineRegistryTest` and also provide a little changelog for the changes you did?

Reply:
Sure - I do it after my vacation next week.

Reply:
@keulinho do you know why this whole check is needed - shopware has the state-machine logic and the code currently in place "hacks" the whole logic so that "cancel" is always possible for every type of state

my fix added the logic to only allow cancel payment (I've added this because this is mentioned in the comment - so this now do the correct thing) but why...

but for me the whole condition should be removed so that the sate machine allows only the configured transitions

Reply:
Hi @0x4 , apologies for the delayed response.

We are having some internal discussions about the point you raised. I will give you feedback as soon as I have some :)

Reply:
Hi @0x4,

Apologies for the long delay in getting back to this PR.

We've had a few chats and we agree with you. This is something that should be fixed in the state machine itself. If you feel up to it, you are welcome to create the state machine migrations to allow this. There are 5 state machines in core that should have this migration:
- Order
- Order Delivery
- Order Transaction
- Order Transaction Capture
- Order Transaction Capture Refund

There is a [state machine migration helper class](https://github.com/shopware/shopware/blob/trunk/src/Core/Migration/Traits/StateMachineMigrationImporter.php) that you can use as well as a [trait](https://github.com/shopware/shopware/blob/trunk/src/Core/Migration/Traits/StateMachineMigrationTrait.php). Here is an [example](https://github.com/shopware/shopware/blob/trunk/src/Core/Migration/V6_4/Migration1643878976AddCaptureRefundStateMachines.php) of how to use the trait.

Please let me know if you are willing to write these migrations. 🙂

Reply:
hi @Bird87ZA 

I'm not sure if I understand it correctly. As far as I see the 5 state machines you mentioned have all the cancel state and cancel transitions for different states. 

My mentioned code in the StateMachineRegistry allowed cancel from every "FROM" state
So you want to add these transitions to every state in every state machine - so that the behaviour is still the same but we can remove this check from the StateMachineRegistry? 

My idea was: remove only the check in the statemachineregistry - as the statemachines itself for me have valid transtions - and there is no need to allow to cancel everything... :-)

Reply:
You are indeed right. The assumption was made that the state machine doesn't handle these states properly.

I had a look at them, and they seem to be doing what they should, which means it is safe to just remove this 'override'.

You are free to do so, then I'll get it merged. :)

Reply:
ok - I have updated the PR

Reply:
Just one comment from the internal review if you don't mind